src/main/java/com/github/manolo8/darkbot/backpage/DispatchManager.java

```diff
@@ -0,0 +1,172 @@
+            if (!m.find()) return false;
+
+            do {
+                int max = Math.min(m.groupCount(), consumers.size());
+                for (int i = 0; i < max; i++)
+                    consumers.get(i).accept(data, m.group(i + 1));
+            } while (m.find());
+
+            return m.groupCount() == consumers.size();
+        }
+
+        public static boolean updateAll(String page, DispatchData data) {
+            boolean updated = true;
+            for (InfoReader reader : InfoReader.values()) {
+                updated &= reader.update(page, data);
+            }
+            return updated;
+        }
+    }
+
+    public boolean hireRetriever(Retriever retriever){
+        if(data.getSlots() > 0){
+            //retriever.getCost(); check cost
+            try {
+                String x = main.backpage.getConnection("ajax/dispatch.php", Method.POST)
+                        .setRawParam("command", "sendDispatch")
+                        .setRawParam("dispatchId", retriever.getId())
+                        .getContent();
+                if(x.contains("ERROR")){
+                    System.out.println("No available dispatch slots");
+                    data.setSlots(0);
+                }else{
+                    System.out.println("Dispatch sent: " + retriever.getId());
+                    data.setSlots(data.getSlots()-1);
+                    return true;
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+                System.out.println("Some error hiring dispatcher: "+ e.toString());
+            }
+        }
+        return false;
+    }
+
+    public String collect(InProgress progress){
+        try {
+            if(progress.getCollectable().equals("0")) return null;
+            System.out.println("Collecting: Slot " + progress.getSlotID());
+            String x = main.backpage.getConnection("ajax/dispatch.php", Method.POST)
+                    .setRawParam("command", "collectDispatch")
+                    .setRawParam("slot", progress.getSlotID())
+                    .getContent();
+
+            //parse response
+            //{"result":"OK","message":"Collected the following:","rewardsLog":[{"lootId":"Solidus","amount":3},{"lootId":"PLT-2026","amount":97},{"lootId":"Scrap","amount":3}]}
+
+            if (x.contains("ERROR")) {
+                System.out.println("Unable to collect retriever");
+            } else {
+                System.out.println("Dispatch Collected: " + progress.getSlotID() + " : " + x.substring(x.indexOf("rewardsLog")));
+                return progress.getName();
+            }
+            progress.setCollectable("0");
+            //remove or do something with slot / empty?
+        } catch (Exception e) {
+            e.printStackTrace();
+            System.out.println("Some error collecting dispatcher: "+ e.toString());
+        }
+        return null;
+    }
+
+    public List<String> collectAll(){
```

Comment:
```suggestion
    public List<String> collectAll() {
```

this should probably be the one that collects all the possible ones, and later calls to refresh